packages/ui/app/src/services/useSearchService.ts

```diff
@@ -0,0 +1,34 @@
+import algolia, { type SearchClient } from "algoliasearch/lite";
+import { useDocsContext } from "../docs-context/useDocsContext";
+
+export type SearchService =
+    | {
+          isAvailable: true;
+          client: SearchClient;
+          index: string;
+      }
+    | {
+          isAvailable: false;
+      };
+
+export function useSearchService(): SearchService {
+    const { docsDefinition } = useDocsContext();
+    const { algoliaSearchIndex } = docsDefinition;
+
+    if (!algoliaSearchIndex) {
+        return { isAvailable: false };
+    }
+
+    if (!process.env.NEXT_PUBLIC_ALGOLIA_APP_ID || !process.env.NEXT_PUBLIC_ALGOLIA_API_KEY) {
+        // TODO: Move this validation elsewhere
+        throw new Error("Missing Algolia variables.");
+    }
+
+    const client = algolia(process.env.NEXT_PUBLIC_ALGOLIA_APP_ID, process.env.NEXT_PUBLIC_ALGOLIA_API_KEY);
```

Comment:
this is gonna run on every render, can we bring it outside the hook?

```ts
if (!process.env.NEXT_PUBLIC_ALGOLIA_APP_ID || !process.env.NEXT_PUBLIC_ALGOLIA_API_KEY) {
  // TODO: Move this validation elsewhere
  throw new Error("Missing Algolia variables.");
}

const client = algolia(process.env.NEXT_PUBLIC_ALGOLIA_APP_ID, process.env.NEXT_PUBLIC_ALGOLIA_API_KEY);

export function useSearchService(): SearchService {
  ...
}
```

Reply:
Right, I should have used useMemo here. But sure, can take it out of the function.

Reply:
Right, I should have used useMemo here. But sure, can take it out of the function.

Reply:
dope sg. benefit of outside the hook is that it's one shared instance for everyone. With `useMemo` every component that calls `useSearchService` will create its own instance